core/src/main/groovy/noe/rhel/server/httpd/HttpdBaseOS.groovy

```diff
@@ -0,0 +1,119 @@
+package noe.rhel.server.httpd
+
+import groovy.util.logging.Slf4j
+import noe.common.utils.Cmd
+import noe.common.utils.JBFile
+import noe.common.utils.Library
+import noe.common.utils.Version
+import noe.server.Httpd
+/**
+ * RHEL BaseOS Httpd server
+ * @author Jan Onderka <jonderka@redhat.com> *
+ */
+@Slf4j
+class HttpdBaseOS extends Httpd {
+  protected String serviceName
+
+  HttpdBaseOS(String basedir, version) {
+    super(basedir, version)
+    this.serviceName = "httpd"
+    this.basedir = "/etc/${serviceName}"
+
+    // please initialize in setDefault() method
+    setDefault()
+    logDirs.each { JBFile.makeAccessible(new File(getServerRoot() + it)) }
+  }
+
+  void setDefault() {
+    super.setDefault()
+    this.configDirs = ['/conf', '/conf.d']
+    if (version >= new Version("2.4")) {
+      this.configDirs.add('/conf.modules.d')
+    }
+    this.modulesDir = this.basedir + "/modules"
+    this.logDirs = ['/logs']
+    // what path will be taken as relative for star/stop cmd
+    this.binPath = '/'
+    if (platform.isRHEL6() || platform.isRHEL5() || platform.isRHEL4()) {
+      this.start = ['service', serviceName, 'start']  // starting as service
+      this.stop = "service ${serviceName} stop"  // stopping as service
+    } else {
+      this.start = ['systemctl', 'start', serviceName]
+      this.stop = ['systemctl', 'stop', serviceName]
+    }
+    this.apachectl = ['/usr/sbin/apachectl']
+    this.deploymentPath = "/var/www/html"
+    this.confDeploymentPath = this.basedir + '/conf.d'
+    this.cgiDeploymentPath = "/var/www/cgi-bin"
+    this.modClusterCacheDir = "/var/cache/mod_cluster"
+    this.opensslPath = 'openssl' // openssl is on $PATH on RHEL by OS installation
+    this.abPath = "/usr/bin/ab"
+    this.htdbmPath = "/usr/bin/htdbm"
+    this.htpasswdPath = "/usr/bin/htpasswd"
+    this.httxt2dbmPath = "/usr/bin/httxt2dbm"
+    this.logresolvePath = "/usr/bin/logresolve"
+
+    // ! please add new paths as absolute paths
+  }
+
+  File getPidFile() {
+    return new File("/var/run/${serviceName}/${serviceName}.pid")
+  }
+
+  Integer extractPid() {
+    def pidAsStr = new ByteArrayOutputStream()
+    if (Cmd.executeCommandRedirectIO("sudo cat ${getPidFile()}", null, null, pidAsStr, System.err) == 0) {
+      pidAsStr = pidAsStr.toString()
+      pidAsStr = pidAsStr.trim()
+      pidAsStr = pidAsStr.replaceAll('"', '')
+      pid = Integer.valueOf(pidAsStr)
+
+      return pid
+    } else {
+      log.debug("PID file is not accessible. B start(ut continuing ...")
+      return null
+    }
+  }
+
+  /**
+   * Stop the server
+   */
+  void start(Map conf = [:]) {
+    super.start(conf)
+    Library.letsSleep(5000)
```

Comment:
@jstefl what do you think of these sleeps? Necessary? I'm inclined to advise on deleting them; or replace them with a loop that checks whether server is up via get requests with some timeout. 

A hard 5s timeout for each server start seems like a significant performance hit to me (+ 3s for killing it), since the server is started and stopped before and after each test (and several times within some tests)

Reply:
Removing as it is superfluous and not needed any more as `ServerAbstract()` already contains `waitForStartComplete()

Reply:
I don't like the hard coded timeouts either, but I remember we were experiencing some issues on slow machines, when we needed to start and stop immediately. I _think_ problem was that the server was not started|stopped exactly at the moment when port were opened (or what do we check). Correct way would be _probably_ to add check that server logs contain message that server started or so (not sure, whether httpd has such message), or work with logs in other 'clever' way :).

I am afraid, this could be non-trivial to solve as rock solid, so I would skip that in this MR. Plain remove could bring some headaches in a future, I don't recommend that.

Reply:
Ok, then. I'll return `start()` and `stop()` methods back with sleep time.

Reply:
If that's the case, this might be quite a pertinent issue to solve. 8s delay for each test will kill us is pretty horrible perf hit if you ask me.

Reply:
@jan-onderka How many tests we are about to run with `HttpdBaseOS` ?

Reply:
There should be only few modules (mod_cluster, mod_jk, ?mod_proxy), so probably higher decades, currently mod_cluster 50, mod_jk 5 (7,5 minute generated delay). I don't expect higher hundreds at all.

Reply:
Ok, it is not nice, but it is not real problem for execution. I don't insist on solution of this in this MR.